Userland/Libraries/LibJS/Interpreter.cpp

```diff
@@ -140,6 +140,7 @@ void Interpreter::exit_scope(const ScopeNode& scope_node)
 
 void Interpreter::enter_node(const ASTNode& node)
 {
+    vm().call_frame().source_range = node.source_range();
```

Comment:
why do we need to store the current source range on the call frame - we already store the node on the VM, can't we get it from there?
Also, the name is probably not ideal, since this isn't the "call frame's source range" but merely the current AST node's source range stored on the current call frame.

Reply:
I'm not familiar with LibJS internals, but the call stack and the node stack are two independent data structures. It's not obvious to me that an AST node is guaranteed to outlive a particular call frame unless it takes an `AK::RefPtr` to it, which would be very expensive here right inside the interpreter loop...

I'd be happy to stick a raw pointer to an AST node here instead, but only if I'm explicitly told it's OK to do that.

Reply:
> It's not obvious to me that an AST node is guaranteed to outlive a particular call frame

They will, they're all children of `JS::Program` which outlives execution of itself.

What I meant is, we already push/pop AST nodes as they are executed, and you can reach the current one via `vm().node_stack().last()`, but I now realise why that on its own with no link to individual call frames is insufficient.

FWIW it's a `Vector<const ASTNode*>` though so yes, that's ok. This is really cool btw :)